crates/fluvio-hub-util/src/fvm/api.rs

```diff
@@ -0,0 +1,116 @@
+//! Hub FVM API Client
+
+use anyhow::{Error, Result};
+use surf::get;
+use url::Url;
+
+use super::{Channel, PackageSet, RustTarget};
+
+/// Base URL for `fmv` related API requests
+pub const API_BASE_URL: &str = "https://hub.infinyon.cloud/hub/v1/fvm";
+
+/// HTTP Client for interacting with the Hub FVM API
+pub struct Client {
+    api_url: Url,
+}
+
+impl Default for Client {
+    fn default() -> Self {
+        // Safe to `unwrap`, `API_BASE_URL` is a valid URL and it's a constant
+        Self::new(Url::parse(API_BASE_URL).unwrap())
+    }
+}
+
+impl Client {
+    pub fn new(api_url: Url) -> Self {
+        Self { api_url }
+    }
+
+    /// Fetches a [`PackageSet`] from the Hub with the specific [`Channel`]
+    #[allow(unused)]
+    pub async fn fetch_package_set<S: AsRef<str>>(
+        &self,
+        name: S,
+        channel: &Channel,
+        arch: &RustTarget,
+    ) -> Result<PackageSet> {
+        let url = self.make_fetch_package_set_url(name, channel, arch)?;
+        let mut res = get(url).await.map_err(|err| {
+            tracing::error!("Failed to fetch package set: {}", err);
```

Comment:
remove redudant error

Reply:
Here I map it to `anyhow::Error` because `surf::Error` doesn't implement `StdError` so we cant use the `?` operator straight.

Reply:
I meant, remove `tracing:error` since it will be converted to upstream error

Reply:
Oh! Of course!